rasa/core/nlg/response.py

```diff
@@ -36,8 +36,9 @@ def _matches_filled_slots(
             value = constraint["value"]
             filled_slots_value = filled_slots.get(name)
             if isinstance(filled_slots_value, str) and isinstance(value, str):
-                return filled_slots_value.casefold() == value.casefold()
-            if filled_slots.get(name) != value:
+                if filled_slots_value.casefold() != value.casefold():
+                    return False
+            elif filled_slots.get(name) != value:
```

Comment:
Tiny adjustment: you can reuse the `filled_slots_value` variable here instead of fetching the value from dict again.